content/intro-to-reading-data.md

```diff
@@ -1,148 +1,120 @@
+```ts
+import { Connection, PublicKey, clusterApiUrl } from "@solana/web3.js";
 
-`npm install @solana/web3.js`.
+const connection = new Connection(clusterApiUrl("devnet"));
+const address = new PublicKey('CenYq6bDRB7p73EjsPEpiYN7uveyPUTdXkDkgUduboaN');
+const balance = await connection.getBalance(address);
 
-### Connect to the Network
+console.log(`The balance of the account at ${address} is ${balance} lamports`); 
+console.log(`✅ Finished!`)
+```
 
-Every interaction with the Solana network using `@solana/web3.js` is going to happen through a `Connection` object. This object establishes a JSON-RPC connection with a Solana cluster (more on clusters later). For now, we’re going to use the url for the Devnet cluster rather than Mainnet. As the name suggests, this cluster is designed for developer use and testing.
+Then run `npx esrun runme.ts`. 
 
-```tsx
-const connection = new Connection(clusterApiUrl('devnet'));
-```
+The balance returned is in *lamports*. A lamport is the minor unit for Sol, like cents is to US Dollars, or pence is to British pounds. A single lamport represents 0.000000001 SOL. Most of the time when dealing with SOL the system will use lamports instead of SOL. Web3.js provides the constant `LAMPORTS_PER_SOL` for making quick conversions.
 
-### Read from the Network
+```ts
+import { Connection, PublicKey, clusterApiUrl, LAMPORTS_PER_SOL } from "@solana/web3.js";
 
-Once you have a `Connection` object, querying the network is as simple as calling the appropriate methods. For example, to get the balance of a particular address, you do the following:
+const connection = new Connection(clusterApiUrl("devnet"));
+const address = new PublicKey('CenYq6bDRB7p73EjsPEpiYN7uveyPUTdXkDkgUduboaN');
+const balance = await connection.getBalance(address);
+const balanceInSol = balance / LAMPORTS_PER_SOL;
 
-```tsx
-async function getBalanceUsingWeb3(address: PublicKey): Promise<number> {
-    const connection = new Connection(clusterApiUrl('devnet'));
-    return connection.getBalance(address);
-}
+console.log(`The balance of the account at ${address} is ${balanceInSol} SOL`); 
+console.log(`✅ Finished!`)
 ```
 
-The balance returned is in fractional SOL called lamports. A single lamport represents 0.000000001 SOL. Most of the time when dealing with SOL the system will use lamports instead of SOL. Web3.js provides the constant `LAMPORTS_PER_SOL` for making quick conversions.
+Then run `npx esrun check-balance.ts`. You should now see something like:
+
+```
+The balance of the account at CenYq6bDRB7p73EjsPEpiYN7uveyPUTdXkDkgUduboaN is 0.00114144 SOL
+✅ Finished!
+```
 
-...and just like that, now you know how to read data from the Solana blockchain! Once we get into custom data things will get more complicated. But for now, let’s practice what we’ve learned so far.
+...and just like that, you are reading data from the Solana blockchain! 
 
-# Demo
+## Using web3.js in a web app
```

Comment:
The Overview, Demo, and Challenge titles are meant to be present on every lesson. It's a structural header. If you'd like to add information, I suggest we have it as a subheading. I might be open to Demo: <More heading here>, but if that's the case I'd want to go through and adjust all the lessons to match.